dash/App/Tokens/AddToken/index.js

```diff
@@ -8,28 +9,32 @@ class AddToken extends Component {
 
     this.nameDefault = 'Token Name'
     this.symbolDefault = 'SYMBOL'
-    this.chainIdDefault = 'ID'
+    this.chainIdDefault = 1
     this.decimalsDefault = '?'
     this.addressDefault = 'Contract Address'
     this.logoURIDefault = 'Logo URI'
 
     this.req = props.req || {}
     this.token = this.req.token || {}
+    this.activeChains = props.activeChains
 
     const chainId = parseInt(this.token.chainId)
     const decimals = parseInt(this.token.decimals)
 
     this.state = {
       name: this.token.name || this.nameDefault,
       symbol: (this.token.symbol || '').toUpperCase() || this.symbolDefault,
-      chainId: (Number.isInteger(chainId) && chainId) || props.currentNetworkId || this.chainIdDefault,
+      chainId: (Number.isInteger(chainId) && chainId) || this.chainIdDefault,
       address: (this.token.address || '').toLowerCase() || this.addressDefault,
       decimals: (Number.isInteger(decimals) && decimals) || this.decimalsDefault,
       logoURI: this.token.logoURI || this.logoURIDefault
     }
   }
 
   async updateTokenData (contractAddress, chainId) {
+    if (contractAddress === this.addressDefault || !chainId) {
```

Comment:
just a matter of style but I prefer to do these kind of checks outside the method (in the calling code) so that we can always assume that the method "does what it says it does" instead of sometimes silently no-oping without letting the calling code know. up to you though

Reply:
I agree actually, it's a code smell at least